src/api/providers/bedrock.ts

```diff
@@ -119,7 +119,14 @@ export class AwsBedrockHandler implements ApiHandler {
 	}
 
 	private async getModelId(): Promise<string> {
-		if (this.options.awsUseCrossRegionInference) {
+
+	// Automatically enable cross-region inference for Claude 3.7 Sonnet
+	if (this.getModel().id === "anthropic.claude-3-7-sonnet-20250219-v1:0") {
+		this.options.awsUseCrossRegionInference = true;
+		return `us.${this.getModel().id}`
```

Comment:
Instead of returning here, we can go into the root of if (this.options.awsUseCrossRegionInference) { in the next process to ensure consistency even when a non-us- prefix comes in.

Reply:
and where would that be ?